pipeline-runner/R/gem2s-5-create_seurat.R

```diff
@@ -1,132 +1,129 @@
-#  - Create a seurat object per sample
-#  - Adding emtpyDrops,  scrublet and MT-content
-#  - Create a file with flag_filtered
-
-# LOADING SPARSE MATRIX AND CONFIGURATION
-create_seurat <- function(input, pipeline_config) {
-  message("reloading old matrices...")
-  scdata_list <- readRDS("/output/pre-doublet-scdata_list.rds")
-
-  message("Loading configuration...")
-  config <- RJSONIO::fromJSON("/input/meta.json")
-
-  print_config(5, "Create Seurat", input, pipeline_config, config)
-
-  # Check which samples have been selected. Otherwiser we are going to use all of them.
-  if (length(config$samples) > 0) {
-    samples <- config$samples
-  } else {
-    samples <- names(scdata_list)
+#' Create a SeuratObject per sample
+#'
+#' @inheritParams download_cellranger
+#' @param prev_out  'output' slot from call to \code{score_doublets}
+#'
+#' @return \code{prev_out} with added slot 'scdata_list' containing \code{SeuratObject}'s for each sample.
+#'
+#' @export
+#'
+create_seurat <- function(input, pipeline_config, prev_out) {
+
+  message("Creating Seurat Objects...")
+  message("items in prev_out: ", paste(names(prev_out), collapse = ' - '))
+
+  # destructure previous output
+  list2env(prev_out, envir = environment())
+
+  samples <- names(counts_list)
```

Comment:
where did this `counts_list` come from? I don't see it in the function declaration, or anywhere else

Reply:
Its from the `list2env` call. There were several things to take from `prev_out` so this simplifies destructuring but I can make it all more explicit

Reply:
> There is a file called `gem2s-5-test_object.R` that is put under folder `R`, it will be better to move it to be under the tests folder

Yes @ogibson-biomage created it as a test of the `SeuratObject` before completing gem2s but then disabled it. Is it okay to remove @ogibson-biomage or are we going to restore it's functionality?

Reply:
I think it is fine as it is now with the comment on top. Just again, make sure you are checking that everything that is compulsory is in prev_out